core/modules/filter/filter.module

```diff
@@ -1585,6 +1585,62 @@ function filter_dom_serialize_escape_cdata_element($dom_document, $dom_element,
 }
 
 /**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function filter_form_filter_format_editor_image_form_alter(&$form, &$form_state) {
+  // Mark files uploaded on entities as temporary, as they will be updated and
+  // added into the file usage table in filter_entity_insert().
+  $entity = _filter_page_is_entity();
+  if (is_object($entity)) {
+    $form['image']['fid']['#temporary'] = TRUE;
+  }
+}
```

Comment:
We don't need to alter the form, Filter module already provides this form itself, so just put this code directly in `filter_format_editor_image_form()`.